### PR TITLE
Check and filter private host addresses

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
- By default forbid host addresses from a private address space.
- Filter private (and invalid) addresses from Kademlia lookup responses.
- Check on startup if the `--host` option and configuration are valid addresses.
- The address `0.0.0.0` (all IPv4 addresses on the local machine) is not a valid `--host` option.
- Introduce a new option `--allow-private-addresses`.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3435

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

**Comment by adaszko** [comment]
`The address 0.0.0.0 (all IPv4 addresses on the local machine) is not a valid --host option.`

Why?  I think it should be.

**Comment by sebtomba** [comment]
@adaszko the purpose of the `--host` option is to specify manually an exact hostname or address. The `0.0.0.0` address is the default behavior of the node if the `--host` option/configuration is not provided. The node guesses then the host address. On the other hand, a node listens on all network interfaces in either case. But it needs to communicate to other nodes only one address. So `all IPv4 addresses` is not an option here.

File: comm/src/main/scala/coop/rchain/comm/package.scala [hunk 1]
```diff
@@ -12,4 +17,20 @@ package object comm {
   }
 
   val CommMetricsSource: Metrics.Source = Metrics.Source(Metrics.BaseSource, "comm")
+
+  def isValidInetAddress(host: String): Boolean =
+    Try(InetAddress.getByName(host))
```
Comment (lukasz-golebiewski) [inline comm/src/main/scala/coop/rchain/comm/package.scala]: I don't think `InetAddress.getByName` is a pure function, so it should be enclosed in `F` and delayed

**Comment by adaszko** [comment]
> The 0.0.0.0 address is the default behavior of the node if the --host option/configuration is not provided. The node guesses then the host address.

The semantics of 0.0.0.0 aren't to guess an address but to use all addresses.

File: comm/src/main/scala/coop/rchain/comm/package.scala (hunk 1 above)
Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/package.scala]: @lukasz-golebiewski done

**Pull request merged**